### PR TITLE
feat: update registration API with new camper model changes

### PR DESCRIPTION
**Pull request by udyding**
## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Tasks-38b49f83f6ce4b07ae188dabb37631e8?p=4885ea4ec8f04377ab6268b2d3a0ee6d)

## Implementation description
* Updated the `camperValidator` in correspondence to new updates to the camper model and DTO
* Updated the camper service - camper creation, get campers 

## Steps to test
1. Test creating a camper with the new camper DTO structure, ensure that camper is created in the database with no errors. Also, ensure that `formResponses` is formatted correctly

## What should reviewers focus on?
* No type errors that arise from the new model changes.
* I modified the interface and schema definition for the camper for just the `formResponses` as I was getting type errors, but now it is fully functional. Simply had to define `formResponses` in the interface as an object instead of Mixed type

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/middlewares/validators/util.ts [hunk 1]
```diff
@@ -77,3 +78,21 @@ export const validateUserEmail = (userEmail: string): boolean => {
     userEmail.split("@")[1] === "uwblueprint.org"
   );
 };
+
+export const validateFormResponses = (
```
Comment (amanda-guo) [inline backend/typescript/middlewares/validators/util.ts:82]: maybe we can generalize this to a `validateMap` type of function? bc we're changing `formResponses` to map so we could use a generic validator in that case

File: backend/typescript/middlewares/validators/camperValidators.ts [hunk 2]
```diff
@@ -8,53 +13,16 @@ export const createCamperDtoValidator = async (
...
-    return res.status(400).send(getApiValidationError("hasLaptop", "boolean"));
-  }
-  if (req.body.allergies && !validatePrimitive(req.body.allergies, "string")) {
-    return res.status(400).send(getApiValidationError("allergies", "string"));
-  }
   if (
-    req.body.additionalDetails &&
-    !validatePrimitive(req.body.additionalDetails, "string")
+    req.body.formResponses &&
+    !validateFormResponses(req.body.formResponses, "string", "string")
```
Comment (amanda-guo) [inline backend/typescript/middlewares/validators/camperValidators.ts:21]: if changing the function in `util.ts`, then make sure to update the call here!

File: backend/typescript/models/camper.model.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import mongoose, { Schema, Document } from "mongoose";
-import { DropOffType } from "../types";
 
 export interface Camper extends Document {
   id: string;
-  camp: Schema.Types.ObjectId,
-  formResponses: Schema.Types.Mixed;
-  dropOffType: DropOffType;
+  camp: Schema.Types.ObjectId;
+  formResponses: {
```
Comment (amanda-guo) [inline backend/typescript/models/camper.model.ts:6]: remember to update to a map

File: backend/typescript/types.ts [hunk 4]
```diff
@@ -25,19 +25,18 @@ export type FormQuestionDTO = {
...
 export type CamperDTO = {
   id: string;
   camp: string;
-  dropOffType: DropOffType;
   registrationDate: Date;
   hasPaid: boolean;
   chargeId: number;
   formResponses: {
-    [key: string]: string
+    [key: string]: string;
```
Comment (amanda-guo) [inline backend/typescript/types.ts:39]: change to a map

File: backend/typescript/middlewares/validators/util.ts [hunk 5]
```diff
@@ -79,7 +79,7 @@ export const validateUserEmail = (userEmail: string): boolean => {
   );
 };
 
-export const validateFormResponses = (
+export const validateMap = (
   formResponses: any,
```
Comment (annechung) [inline backend/typescript/middlewares/validators/util.ts:83]: nit: should we rename this variable to map as well if we're making this a general validator? @amanda-guo 
```suggestion
  map: any,
```

**Review by annechung — COMMENTED** [review]
whoops didn't realize the commits in this branch went into main already D: just one nit on the remaining changes

**Review by annechung — APPROVED** [review]
looks goood

**Pull request merged**